### ISSUE TITLE
Using a password dump for a corpus

### ISSUE THREAD

**Issue by zachwhalen** [dariusk/NaNoGenMo-2015#4]
I have several ideas and I fully intend to implement one of them by the end of the month. I think the <a href="https://web.archive.org/web/20150316185559/https://xato.net/passwords/ten-million-passwords/">10 million passwords and usernames dump</a> would be an interesting corpus to play with.

**Comment by hugovk** [comment]
> ### The journal of a password cracker
> 
> First I entered youliaha and 3814529 but didn't get it.
> So I typed in LTrain823 and amdathlo but didn't gain access.
> Changing tack, I keyed in aleksey_80afonskiy and afonya0603 but no dice.
> Next I tried 000126 and avocat1 which -- guess what -- was unsuccessful.
> ...
> Finally, I bashed in arsefrem and arsefremenko and surprise surprise! Nothing.
> Oh how I wish they would automate this.

**Comment by MichaelPaulukonis** [comment]
> > Oh how I wish they would automate this.

How could such a thing be automated?

## IT'S AGAINST NATURE!

**Comment by zachwhalen** [comment]
Adding this as it may be of interest:

Last year, I attempted to create a graphic novel. <a href="https://github.com/zachwhalen/NaGraNoGenMo">The result</a> was late and not as compelling as I hoped it would be, _The Something, The Thanksgiving, and The Nothing_. 

I made an off-season sequel that I think is more compelling. So much so I decided to self-publish it with Lulu. Here's the first of what may be a series of blogs explaining <a href="http://www.zachwhalen.net/posts/an-arthrogram/">_An Arthrogram_</a>, which is a book of empty comic layouts. Every possible layout given a constrained _Watchmen_-like grid of 3 x 3 panels.

https://www.youtube.com/embed/JCPj1YXI4-g

**Comment by zachwhalen** [comment]
Just a quick update. I've set up <a href="https://github.com/zachwhalen/secretgardens/">a repository</a> but I'm going back and forth about what I actually want to do. My first idea was to use this password dump to generate a novel that already exists, Pierre Menard-style. 

So in my first version, the script scans the password dump for correctly-spelled English words, then looks for those words in the novel, _The Secret Garden_. Or rather, it scans _The Secret Garden_, and for each word it finds in the passwordlist, it prints that word. Words it fails to find are just printed as *****:

> ***\* Mary *****\* was sent to ************\* Manor to live with her *****\* 
> everybody said she was the most *******************\* child ever ****\* 
> It was true too  She had a little thin face and a little thin ****\* 
> thin light hair and a sour expression  Her hair was yellow and ***\* 
> face was yellow because she had been born in India and had always ****\* 
> ill in one way or another  Her father had held a position under ***\* 
> English *********\* and had always been busy and ill himself and ***\* 
> mother had been a great beauty who ****\* only to go to parties ***\* 
> amuse herself with gay people  She had not wanted a little girl **\* 
> all and when Mary was born she handed her over to the care of an ****\* 
> who was made to understand that if she wished to please the **\* *****\* 
> she must keep the child out of sight as much as possible  So when ***\* 
> was a sickly ******\* ugly little baby she was kept out of the ***\* 
> and when she became a sickly ******\* *******\* thing she was kept ***\* 
> of the way also  She never *********\* seeing *********\* anything ***\* 
> the dark faces of her ***\* and the other native servants and as ****\* 
> always obeyed her and gave her her own way in everything because ***\* 
> **\* ****\* would be angry if she was disturbed by her crying by ***\* 
> time she was six years old she was as *********\* and selfish a ******\* 
> pig as ever lived  The young English governess who came to teach ***\* 
> to read and write *******\* her so much that she gave up her place **\* 
> three months and when other **********\* came to try to fill it ****\* 
> always went away in a shorter time than the first one  So if Mary ***\* 
> not chosen to really want to know how to read books she would *****\* 
> have learned her letters at ***\* 

This is interesting, to me at least, because each word you see here is something that someone used as a password. I can't decide, though, if it's _interesting enough_, so I'm trying to learn how to use POS tagging to do some smarter parsing. I'm thinking perhaps of replacing parts of speech with the frequency matching. Like, replace the most frequently used noun in the book with the most frequently used noun password (that is, "password").

**Comment by MichaelPaulukonis** [comment]
_The **Secret** Garden_

....

I SEE WHAT YOU DID THERE

---

What if the words NOT found as passwords were left, and the secret-passwords were removed from the text-garden instead? 

Not a lot would be left, though.

**Comment by zachwhalen** [comment]
@MichaelPaulukonis Right, you can kind of see how it wouldn't be much left. Mostly proper nouns, I think. 

Still, it might be interesting to do something else with that smaller set of non-overlapping terms.

**Comment by zachwhalen** [comment]
By the way, I'm doing this in Perl because why not and I'm using with the `Lingua::EN::Tagger` package for POS tagging. It seems to work fine.

**Comment by hugovk** [comment]
> So in my first version, the script scans the password dump for correctly-spelled English words, then looks for those words in the novel, The Secret Garden. Or rather, it scans The Secret Garden, and for each word it finds in the passwordlist, it prints that word. Words it fails to find are just printed as *****:

Are "incorrectly-spelled" words left out of the equation? (The "Or rather" suggests this isn't the case.) But if so, I'd leave them it. "Correctly" spelled depends on the authority of the dictionary list, and you might miss some unusual (or misspelt) words in the original that are also passwords.

I'd have the **\* the same length as the omitted words.

I quite like keeping the original text more intact than replacing, say, each "time" with "password". (But try both!) It's readable-ish. If you're generating HTML output you could style each word with some sort of intensity to represent how frequent that word is.

**Comment by zachwhalen** [comment]
> If you're generating HTML output you could style each word with some sort of intensity to represent how frequent that word is.

Interesting idea. I could do something with relative sizing or transparency. Sizing could be interesting since it would inevitable make the layout much more dynamic.

**Comment by hugovk** [comment]
Perhaps also change the background and foreground colours too.
